### PR TITLE
Fix issue #680: Port from host header should be removed only protocol…

### PR DESCRIPTION
**Pull request by kullanici0606**
… matches

### CODE REVIEW

**Comment by kannappanr** [comment]
@kullanici0606 Can you please fix the formatting issues?

File: api/src/main/java/io/minio/MinioClient.java [hunk 1]
```diff
@@ -1049,6 +1049,17 @@ public void writeTo(BufferedSink sink) throws IOException {
...
+  
+  /**
+   * Checks whether port should be omitted in Host header
+   * 
+   * HTTP Spec (rfc2616) defines that port should be omitted in Host header
+   * when port and service matches (i.e HTTP -> 80, HTTPS -> 443)
+   * @param url Url object
+   */
+  private boolean shouldOmitPortInHostHeader(HttpUrl url) {
+	  return (url.scheme().equals("http") && url.port() == 80) || (url.scheme().equals("https") && url.port() == 443);
```
Comment (balamurugana) [inline api/src/main/java/io/minio/MinioClient.java]: Wondering someone is using non-standard port for TLS and non-TLS i.e. for TLS port 80 is used and for non-TLS port 443 is used.

Is this your problem or just making spec compatible?

Comment (kullanici0606) [reply api/src/main/java/io/minio/MinioClient.java]: I also have this problem but the change is making the code spec compatible.

**Comment by kullanici0606** [comment]
I fixed the formatting issues, could you please review it?

File: api/src/main/java/io/minio/MinioClient.java (hunk 1 above)
Comment (balamurugana) [reply api/src/main/java/io/minio/MinioClient.java]: Got it.

**Pull request merged**